packages/nibijs/src/chain/types.ts

```diff
@@ -1,35 +1,23 @@
 import { Coin } from "@cosmjs/proto-signing"
-import { ABCIEvent } from "../tx/event"
 import BigNumber from "bignumber.js"
+import { ABCIEvent } from "../tx/event"
 
-/** go: Generic function for executing asynchronous code safely inside a
- * try-catch by default. If the promise succeeds, it's result is returned and the
- * error is undefined. If it fails, the result is undefined and the error message
- * is present. The function's called "go" because it mimics the error handling
- * behavior from Golang.
+/**
+ * Asserts that a given condition is true. If the condition evaluates to false,
+ * an "AssertionError" is thrown with an optional custom message.
  *
- * GoSuccess<T>: A successful promise result that doesn't "throw".
- * GoError: A wrapped error type resulting from a "throw" inside a Promise.
- * */
-export const go = async <T>(
-  promise: Promise<T>
-): Promise<GoSuccess<T> | GoError> => {
-  try {
-    return { res: await promise, err: undefined }
-  } catch (err) {
-    return { res: undefined, err: (err as Error).message }
-  }
-}
-
-/** GoSuccess: A successful promise result that doesn't "throw". */
-type GoSuccess<T> = { res: T; err: undefined }
-/** GoError: A wrapped error type resulting from a "throw" inside a Promise. */
-type GoError = { res: undefined; err: string }
-
-export const assert = (condition: boolean, message?: string) => {
+ * @param {boolean} condition - The condition to test.
+ * @param {string} message - Optional. A custom error message to display if the
+ * assertion fails.
+ * @returns {boolean} - Returns true if the assertion is successful.
+ */
+export const assert = (
+  condition: boolean,
+  message?: string
+): boolean | string => {
   if (!condition) {
     const errMsg = message ? `AssertionError: ${message}` : "AssertionError"
-    console.error(Error(errMsg))
+    console.error(errMsg)
```

Comment:
With the change of an Error object to a string, does this not break tests?

Also would we prefer it as a string?
Would an actual error object be more useful? (stack trace)

Reply:
Resolved by c08edb604749e90202e4e08ae399ca84f91fa451